.github/workflows/ci.yml

```diff
@@ -79,11 +80,29 @@ jobs:
     if: github.event_name == 'pull_request'
     steps:
       - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0 # checkout all commits to be able to determine merge base
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'zulu'
+          java-version: 17
+          cache: 'maven'
+      - name: Fetch base ref to find merge-base for GIB
+        run: .github/bin/git-fetch-base-ref.sh
       - name: Check Commits
         uses: trinodb/github-actions/block-commits@c2991972560c5219d9ae5fb68c0c9d687ffcdd10
         with:
           action-merge: fail
           action-fixup: none
+      - name: Check if all intermediate commits of this PR compile
+        run: |
+          git config user.name "$(git --no-pager log --format=format:'%an' -n 1)"
+          git config user.email "$(git --no-pager log --format=format:'%ae' -n 1)"
+          git log --graph --oneline --all
+          git rebase --exec "$MAVEN package $MAVEN_COMPILE_COMMITS || (git status && false)" refs/remotes/origin/${{ github.event.pull_request.base.ref }}
```

Comment:
technically speaking `$(git merge-base HEAD refs/remotes/origin/${{ github.event.pull_request.base.ref }} )` is a better rebase target